driver/config/test_config.go

```diff
@@ -51,14 +55,98 @@ func WithConfigValue(ctx context.Context, key string, value any) context.Context
 	return WithConfigValues(ctx, map[string]any{key: value})
 }
 
-func WithConfigValues(ctx context.Context, setValues map[string]any) context.Context {
+func WithConfigValues(ctx context.Context, setValues ...map[string]any) context.Context {
 	values, ok := ctx.Value(contextConfigKey).([]map[string]any)
 	if !ok {
 		values = make([]map[string]any, 0)
 	}
-	newValues := make([]map[string]any, len(values), len(values)+1)
+	newValues := make([]map[string]any, len(values), len(values)+len(setValues))
 	copy(newValues, values)
-	newValues = append(newValues, setValues)
+	newValues = append(newValues, setValues...)
 
 	return context.WithValue(ctx, contextConfigKey, newValues)
 }
+
+type ConfigurableTestHandler struct {
+	configs map[uuid.UUID][]map[string]any
+	handler http.Handler
+}
+
+func NewConfigurableTestHandler(h http.Handler) *ConfigurableTestHandler {
+	return &ConfigurableTestHandler{
+		configs: make(map[uuid.UUID][]map[string]any),
+		handler: h,
+	}
+}
+
+func (t *ConfigurableTestHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	cID := r.Header.Get("Test-Config-Id")
+	if config, ok := t.configs[uuid.FromStringOrNil(cID)]; ok {
+		r = r.WithContext(WithConfigValues(r.Context(), config...))
+	}
```

Comment:
The handler looks up the config ID from the request header and adds the config to the request context server-side if it found it.